### PR TITLE
Add Hints.md  exercises.

### PR DESCRIPTION
**Pull request by GalaDe**

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
First of all, thanks so much for doing this! It's a huge amount of work. I'll do a review somewhere today. 

By the way, usually, making pull requests as small as possible is adviced. In this case, having one PR per hints file would make it far easier for me to review things. Furthermore, having one big PR means that it will only get merged if everything in it is okay. If there is a typo in one hints file, the other hint files also won't be merged. With an individual PR per hints file, this wouldn't be a problem. It is more work for the one creating the PR's (you in this case), so there's the trade-off. For this PR, it's fine to leave it as is, just consider it some general advice (as I hope you'll be wanting to do more PR's :)).

Once again, thanks!

File: exercises/hangman/Hints.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Hints
+This exercise requires to work with events. For more information, see [this page]
```
Comment (ErikSchierboom) [inline exercises/hangman/Hints.md]: "This exercise requires to work with events" should be "This exercise requires you to work with events"

**Comment by ErikSchierboom** [comment]
There are several files in the PR that should be removed:

- .fake/build.fsx_204AE3C7.dll
- .fake/build.fsx_204AE3C7_config.xml
- .fake/build.fsx_204AE3C7_warnings.txt
- .paket/paket.exe
- paket-files/nunit/nunit-transforms/nunit3-junit/nunit3-junit.xslt

For this, you could simply delete the `.fake`, `.paket` and `paket-files` directories. My other review comments will follow.

File: exercises/accumulate/Hints.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+## Hints
+This exercise requires you to process a collection of data. You can simplify your code by using LINQ (Language Integrated Query).
```
Comment (ErikSchierboom) [inline exercises/accumulate/Hints.md]: I would rephrase this to note that this exercise requires you to write an extension method. The link being used is great.

File: exercises/alphametics/Hints.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+## Hints
+- To parse the text, you could try to use the [Sprache](https://github.com/sprache/Sprache/blob/develop/README.md) library. 
+There is also good tutorial located [here] (https://www.thomaslevesque.com/2017/02/23/easy-text-parsing-in-c-with-sprache/).
```
Comment (ErikSchierboom) [inline exercises/alphametics/Hints.md]: "There is also good tutorial located [here]" should be "You can find a good tutorial [here]"

File: exercises/forth/Hints.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+## Hints
+- To parse the text, you could try to use the [Sprache](https://github.com/sprache/Sprache/blob/develop/README.md) library. 
+There is also good tutorial located [here] (https://www.thomaslevesque.com/2017/02/23/easy-text-parsing-in-c-with-sprache/).
```
Comment (ErikSchierboom) [inline exercises/forth/Hints.md]: "There is also good tutorial located [here]" should be "You can find a good tutorial [here]"

File: exercises/sgf-parsing/Hints.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+## Hints
+- To parse the text, you could try to use the [Sprache](https://github.com/sprache/Sprache/blob/develop/README.md) library. 
+There is also good tutorial located [here] (https://www.thomaslevesque.com/2017/02/23/easy-text-parsing-in-c-with-sprache/).
```
Comment (ErikSchierboom) [inline exercises/sgf-parsing/Hints.md]: "There is also good tutorial located [here]" should be "You can find a good tutorial [here]"

File: exercises/twelve-days/Hints.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+## Hints
+## Hints
```
Comment (ErikSchierboom) [inline exercises/twelve-days/Hints.md]: Duplicate line

File: exercises/food-chain/Hints.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+## Hints
+## Hints
```
Comment (ErikSchierboom) [inline exercises/food-chain/Hints.md]: Duplicate line

File: exercises/diffie-hellman/Hints.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+## Hints
+This exercise requires you to perform calculations with the large numbers. It might be a good idea to use [BigIntegers]
```
Comment (ErikSchierboom) [inline exercises/diffie-hellman/Hints.md]: "with the large numbers" should be "on large numbers"

Comment (ErikSchierboom) [inline exercises/diffie-hellman/Hints.md]: "It might be a good idea to use [BigIntegers]" should be "To correctly represent large numbers, the [BigInteger](https://msdn.microsoft.com/en-us/library/system.numerics.biginteger(v=vs.110).aspx) struct is used."

File: exercises/bank-account/Hints.md [hunk 9]
```diff
@@ -0,0 +1,4 @@
+## Hints
+This exercise requires you to handling data related to currency and money. A normal approuch is to use the [Decimal] (https://msdn.microsoft.com/en-US/library/system.decimal.aspx) struct to store currency values. But this only stores the numeric of a currency.
```
Comment (ErikSchierboom) [inline exercises/bank-account/Hints.md]: "But this only stores the numeric of a currency." should be "Note though that you then only store the numeric value of a currency."

File: exercises/bank-account/Hints.md [hunk 10]
```diff
@@ -0,0 +1,4 @@
+## Hints
+This exercise requires you to handling data related to currency and money. A normal approuch is to use the [Decimal] (https://msdn.microsoft.com/en-US/library/system.decimal.aspx) struct to store currency values. But this only stores the numeric of a currency. 
+When you want a full string to display on your user interface, you can use the [Decimal.ToString(IFormatProvider)] (https://msdn.microsoft.com/en-US/library/3ebe5aks.aspx) or use the [Decimal.ToString(String)] (https://msdn.microsoft.com/en-US/library/fzeeb5cd.aspx).
```
Comment (ErikSchierboom) [inline exercises/bank-account/Hints.md]: I would remove this line, as I think it gives away too much information. I'd like people to figure this out themselves :)

File: exercises/simple-linked-list/Hints.md [hunk 11]
```diff
@@ -0,0 +1,4 @@
+## Hints
+This exercise requires you to process a collection of data using linked list data structure. In order to do this, you might find yourselves in need to iterate the collection.
```
Comment (ErikSchierboom) [inline exercises/simple-linked-list/Hints.md]: Actually, people *have* to implement `IEnumerable`, so I would rephrase this to something like: "This exercise requires you to create a linked list data structure which can be iterated. This requires you to implement the `IEnumerable<T>` interface". For more information, see [this page] (https://msdn.microsoft.com/en-us/library/9eekhta0(v=vs.110).aspx)."

File: exercises/wordy/Hints.md [hunk 12]
```diff
@@ -0,0 +1,3 @@
+## Hints
+- To parse the text, you could try to use the [Sprache](https://github.com/sprache/Sprache/blob/develop/README.md) library. 
+There is also good tutorial located [here] (https://www.thomaslevesque.com/2017/02/23/easy-text-parsing-in-c-with-sprache/).
```
Comment (ErikSchierboom) [inline exercises/wordy/Hints.md]: "There is also good tutorial located [here]" should be "You can find a good tutorial [here]"

File: exercises/sublist/Hints.md [hunk 13]
```diff
@@ -0,0 +1,4 @@
+## Hints
+This exercise requires you to compare sets of data. In order to do this, you need to use IComparable Interface.
```
Comment (ErikSchierboom) [inline exercises/sublist/Hints.md]: I would rephrase this to something like: "To be able to compare data, the `IComparable` interface is used."

**Comment by GalaDe** [comment]
Hi @ErikSchierboom! Thank you for your advice. This was my first time of doing this:) I will review your comments and make changes tonight. Sorry again for inconvenience.

**Comment by ErikSchierboom** [comment]
@GalaDe Absolutely no problem! I myself have learned this the exact same way: having a maintainer explain why it is useful to have smaller commits. You should be very pleased with what you have done! I'm really happy with it. There are only some small comments, and that was to be expected as you probably haven't even worked on several exercises :)

File: exercises/bank-account/Hints.md (hunk 10 above)
Comment (GalaDe) [inline exercises/bank-account/Hints.md]: Hi @ErikSchierboom, sorry which line should be removed? This one:

+When you want a full string to display on your user interface, you can use the [Decimal.ToString(IFormatProvider)] (https://msdn.microsoft.com/en-US/library/3ebe5aks.aspx) or use the [Decimal.ToString(String)] (https://msdn.microsoft.com/en-US/library/fzeeb5cd.aspx).

**Comment by GalaDe** [comment]
@ErikSchierboom, all done! Thank you again for your help! Looking forward to do some more fun stuff after my return!

**Comment by GalaDe** [comment]
@ErikSchierboom Great! Thank you!

**Comment by ErikSchierboom** [comment]
@GalaDe It's merged! 🎉 Thanks a lot for the work, it is much appreciated.

**Comment by GalaDe** [comment]
Ehyyy:)))) Thank you again for your help!!!!!!!
